### DISCUSSION TITLE
How to get the physical coordinates in the tile-box?

### DISCUSSION THREAD

**Discussion by Horizonatpku** [AMReX-Codes/amrex#2952] [Q&A]
Hi  everyone,

I want to use the physical coordinates in the tile-box (mfi%tilebox) to participate the calculation.

However, I don't know how to get the physical coordinates in the tile-box and even the grid-space of the tile-box (dx, dy, dz).

I will be appreciate if someone can give me some  help!

Best wishes,

He Yifeng

**Comment by atmyers** [comment]
The information needed to map between index-space and physical space coordinates, including the cell spacing and global domain offset, is stored in the `amrex::Geometry` object. You need to query that. You may find this function in WarpX helpful: https://github.com/ECP-WarpX/WarpX/blob/development/Source/WarpX.cpp#L2296

**Comment by Horizonatpku** [comment]
Hi Myers,
 
Thank you for you nice advice!

I am not familiar with AMRex,  I do not know if I understand the code you point out

For example in Fortran codes, I get a tilebox  named tbx and it is located in the level named lev

First, I announce geom :
--------------------------------------
type(amrex_geometry) :: geom
--------------------------------------

then get the specific geom for tbx
--------------------------------------
gm = GetInstance().Geom(lev)
--------------------------------------

Finally, I can get the cell spacing and physical lower bounds of tbx
--------------------------------------
dx=gm.CellSize() 
lowbd=gm.ProbLo()
--------------------------------------

I will be appreciate if you can give me some suggestions, thank u again!

Best wishes,
He Yifeng

**Reply by atmyers** [reply]
If you are using the Fortran Interfaces, you'll need to use the methods here instead of the C++ versions: https://github.com/AMReX-Codes/amrex/blob/development/Src/F_Interfaces/Base/AMReX_geometry_mod.F90

In that case, there is also a `geom%get_physical_location` method that you could use on `tbx%lo` and `tbx%hi`.
